### PR TITLE
fix(restore): use backup from any broker

### LINKED ISSUE #14496: Restore with fewer brokers fails to find all backups
**Describe the bug**

Restoring a backup taken with `clusterSize=3` from a broker with `clusterSize=1` fails because not all backups can be found.

The `PartitionRestoreService`  builds known broker ids based on `clusterSize` and then attempts to find a partition backup by going through all broker ids. When `clusterSize=1`, it will only search for backups taken by broker 1, thus missing backups taken by broker 2 and 3.

**To Reproduce**

1. Take a backup with `clusterSize=3` where broker 1 is _not_ leader for all partitions
2. Restore with `clusterSize=1`

**Expected behavior**

Restore finds backups from all brokers, not only those in the current configuration.

**Context**
Support case: https://jira.camunda.com/browse/SUPPORT-18562

### PR DESCRIPTION
**Pull request by lenaschoenburg**
This fixes #14496 by searching for matching backups taken by any broker instead of  enumerating possible broker ids based on the current configuration.

### CODE REVIEW

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/ClusteredBackupRestoreTest.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            .withPartitionsCount(3)
+            .withReplicationFactor(1)
+            .withBrokerConfig(broker -> configureBackupStore(broker.brokerConfig()))
+            .build()
+            .start()
+            .awaitCompleteTopology()) {
+      final var actuator = BackupActuator.ofAddress(cluster.availableGateway().monitoringAddress());
+
+      try (final var client = cluster.newClientBuilder().build()) {
+        IntStream.range(0, 30)
```
Comment (deepthidevaki) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/ClusteredBackupRestoreTest.java:74]: :question: Do you have to publish 30 messages? May be reduce it to speed up the test.

**Review by deepthidevaki — APPROVED** [review]
:rocket: Nice find.

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/ClusteredBackupRestoreTest.java (hunk 1 above)
Comment (oleschoenburg) [reply qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/ClusteredBackupRestoreTest.java:74]: Not really but it also doesn't have an impact on test time.

**Comment by oleschoenburg** [comment]
bors merge

**Pull request merged**